File: CS/Serialization/MyGridView.cs
using DevExpress.Utils;
using DevExpress.Utils.Serializing;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using System;

namespace Serialization
{
	public delegate void AllowSerializePropertyEventHandler(object sender, AllowSerializePropertyEventArgs e);
	public delegate void SerializeCustomPropertiesEventHandler(object sender, SerializeCustomPropertiesEventArgs e);
	public delegate void CustomPropertyValueEventHandler(object sender, CustomPropertyValueEventArgs e);

	public class MyGridView : GridView
	{
		public MyGridView(GridControl ownerGrid)
			: base(ownerGrid)
		{
		}
		public MyGridView()
			: base()
		{
		}

		protected override string ViewName
		{
			get
			{
				return "MyGridView";
			}
		}

		protected override void SaveLayoutCore(XtraSerializer serializer, object path, OptionsLayoutBase options)
		{
			if ( serializer is XmlXtraSerializer )
				serializer = new MyXmlXtraSerializer();

			base.SaveLayoutCore(serializer, path, options);
		}

		protected override void RestoreLayoutCore(XtraSerializer serializer, object path, OptionsLayoutBase options)
		{
			if ( serializer is XmlXtraSerializer )
				serializer = new MyXmlXtraSerializer();

			base.RestoreLayoutCore(serializer, path, options);
		}

		public event AllowSerializePropertyEventHandler AllowSerializeProperty;
		protected internal virtual void RaiseAllowSerialzeProperty(AllowSerializePropertyEventArgs args)
		{
			if ( AllowSerializeProperty != null )
				AllowSerializeProperty(this, args);
		}

		public event SerializeCustomPropertiesEventHandler SerializeCustomProperties;
		protected internal virtual void RaiseSerializeCustomProperties(SerializeCustomPropertiesEventArgs args)
		{
			if ( SerializeCustomProperties != null )
				SerializeCustomProperties(this, args);
		}

		public event SerializeCustomPropertiesEventHandler DeserializeCustomProperties;
		protected internal virtual void RaiseDeserializeCustomProperties(SerializeCustomPropertiesEventArgs args)
		{
			if ( DeserializeCustomProperties != null )
				DeserializeCustomProperties(this, args);
		}

		public event CustomPropertyValueEventHandler SetCustomPropertyValue;
		protected internal virtual void RaiseSetCustomPropertyValue(CustomPropertyValueEventArgs args)
		{
			if ( SetCustomPropertyValue != null )
				SetCustomPropertyValue(this, args);
		}

		public event CustomPropertyValueEventHandler GetCustomPropertyValue;
		protected internal virtual void RaiseGetCustomPropertyValue(CustomPropertyValueEventArgs args)
		{
			if ( GetCustomPropertyValue != null )
				GetCustomPropertyValue(this, args);
		}
	}
}

File: VB/Serialization/MyGridView.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Utils
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports System

Namespace Serialization
	Public Delegate Sub AllowSerializePropertyEventHandler(ByVal sender As Object, ByVal e As AllowSerializePropertyEventArgs)
	Public Delegate Sub SerializeCustomPropertiesEventHandler(ByVal sender As Object, ByVal e As SerializeCustomPropertiesEventArgs)
	Public Delegate Sub CustomPropertyValueEventHandler(ByVal sender As Object, ByVal e As CustomPropertyValueEventArgs)

	Public Class MyGridView
		Inherits GridView
		Public Sub New(ByVal ownerGrid As GridControl)
			MyBase.New(ownerGrid)
		End Sub
		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides ReadOnly Property ViewName() As String
			Get
				Return "MyGridView"
			End Get
		End Property

		Protected Overrides Sub SaveLayoutCore(ByVal serializer As XtraSerializer, ByVal path As Object, ByVal options As OptionsLayoutBase)
			If TypeOf serializer Is XmlXtraSerializer Then
				serializer = New MyXmlXtraSerializer()
			End If

			MyBase.SaveLayoutCore(serializer, path, options)
		End Sub

		Protected Overrides Sub RestoreLayoutCore(ByVal serializer As XtraSerializer, ByVal path As Object, ByVal options As OptionsLayoutBase)
			If TypeOf serializer Is XmlXtraSerializer Then
				serializer = New MyXmlXtraSerializer()
			End If

			MyBase.RestoreLayoutCore(serializer, path, options)
		End Sub

		Public Event AllowSerializeProperty As AllowSerializePropertyEventHandler
		Protected Friend Overridable Sub RaiseAllowSerialzeProperty(ByVal args As AllowSerializePropertyEventArgs)
			RaiseEvent AllowSerializeProperty(Me, args)
		End Sub

		Public Event SerializeCustomProperties As SerializeCustomPropertiesEventHandler
		Protected Friend Overridable Sub RaiseSerializeCustomProperties(ByVal args As SerializeCustomPropertiesEventArgs)
			RaiseEvent SerializeCustomProperties(Me, args)
		End Sub

		Public Event DeserializeCustomProperties As SerializeCustomPropertiesEventHandler
		Protected Friend Overridable Sub RaiseDeserializeCustomProperties(ByVal args As SerializeCustomPropertiesEventArgs)
			RaiseEvent DeserializeCustomProperties(Me, args)
		End Sub

		Public Event SetCustomPropertyValue As CustomPropertyValueEventHandler
		Protected Friend Overridable Sub RaiseSetCustomPropertyValue(ByVal args As CustomPropertyValueEventArgs)
			RaiseEvent SetCustomPropertyValue(Me, args)
		End Sub

		Public Event GetCustomPropertyValue As CustomPropertyValueEventHandler
		Protected Friend Overridable Sub RaiseGetCustomPropertyValue(ByVal args As CustomPropertyValueEventArgs)
			RaiseEvent GetCustomPropertyValue(Me, args)
		End Sub
	End Class
End Namespace
